src/path_manager.c

```diff
@@ -988,6 +986,22 @@ static void complete_pm_init(struct mptcpd_pm *pm)
          *       ID manager state can be synchronized with the kernel
          *       dynamically.
          */
+
+        /**
+         * @bug Mptcpd plugins should only be loaded once at process
+         *      start.  The @c mptcpd_plugin_load() function only
+         *      loads the functions once, and only reloads after
+         *      @c mptcpd_plugin_unload() is called.
+         */
+        if (!mptcpd_plugin_load(pm->config->plugin_dir,
+                                pm->config->default_plugin,
+                                pm)) {
+                l_error("Unable to load path manager plugins.");
+
+                mptcpd_pm_destroy(pm);
+
+                exit(EXIT_FAILURE);
```

Comment:
I ran a quick leak check through `valgrind`, and found a small memory leak at process exit when this condition (plugin load failure) is triggered:
```
==11756== 40 bytes in 1 blocks are definitely lost in loss record 19 of 25
==11756==    at 0x4839809: malloc (vg_replace_malloc.c:307)
==11756==    by 0x48608FD: l_malloc (util.c:62)
==11756==    by 0x4865D26: watch_add (main.c:144)
==11756==    by 0x4867158: l_io_new (io.c:191)
==11756==    by 0x486BC54: l_genl_new (genl.c:1140)
==11756==    by 0x10D59F: mptcpd_pm_create (path_manager.c:1166)
==11756==    by 0x10B62E: main (mptcpd.c:58)
```
I'll create a separate pull request for the fix.